### PR TITLE
Fix for the infinite loop due to bad comparer

### PR DESCRIPTION
**Pull request by cristinamanum**
Fix for the infinite loop due to bad comparer. 
With this change add the equality in the comparer and add a thread safe random.

### CODE REVIEW

File: src/NuGet.Services.Search.Client/Client/RetryingHttpClientWrapper.cs [hunk 1]
```diff
@@ -212,27 +211,31 @@ private static bool ShouldTryOther(HttpResponseMessage response)
...
-        
-        class WeightedRandomComparer
+
+        public class WeightedRandomComparer
             : IComparer<int>
         {
-            private readonly Random _random;
-
-            public WeightedRandomComparer(Random random)
+            public WeightedRandomComparer()
```
Comment (skofman1) [inline src/NuGet.Services.Search.Client/Client/RetryingHttpClientWrapper.cs]: you can remove the constructor

Comment (cristinamanum) [reply src/NuGet.Services.Search.Client/Client/RetryingHttpClientWrapper.cs]: Updated.

File: src/NuGet.Services.Search.Client/Client/ThreadSafeRandom.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+{
+    /// <summary>
+    /// A thread safe random implementation 
+    /// https://blogs.msdn.microsoft.com/pfxteam/2009/02/19/getting-random-numbers-in-a-thread-safe-way/
+    /// </summary>
+    public static class ThreadSafeRandom
+    {
+        private static Random _global = new Random();
+        [ThreadStatic]
+        private static Random _local;
```
Comment (skofman1) [inline src/NuGet.Services.Search.Client/Client/ThreadSafeRandom.cs]: can you edit the code to comply with our coding style? I mean line breaks, curly brackets around lock statement. etc.

Comment (cristinamanum) [reply src/NuGet.Services.Search.Client/Client/ThreadSafeRandom.cs]: Updated.

**Comment by skofman1** [comment]
Fixes: https://github.com/NuGet/NuGetGallery/issues/5703

**Pull request merged**